winnery_ticket_experiment.py

```diff
@@ -8,14 +8,15 @@ class ZeroWeightsCallback(Callback):
     This is equivalent to removing the given weights.
     """
 
-    def __init__(self, zero_weight_mask):
+    def __init__(self, zero_weight_mask, layer_indices):
         super(ZeroWeightsCallback, self).__init__()
         self.zero_weight_mask = zero_weight_mask
+        self.layer_indices
 
     def _zero_masked_weights(self):
-        for i in range(1, len(self.model.layers)):
+        for i, layer in enumerate(self.layer_indices):
```

Comment:
You have `layer` iterating `1,...,len(self.model.layers)` but `i` iterates from `0,...,len(self.model.layers)-1`, so `self.model.layers[i].get_weights()` is a mistake (what if `i=0`?). What you want here is `for i in layer_indices`